src/components/ContactsView.js

```diff
@@ -1,23 +1,38 @@
-import { useEffect, useState } from "react"
-import { useParams } from "react-router-dom"
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
 
 function ContactsView() {
-  const [contact, setContact] = useState(false)
+  const [contact, setContact] = useState(false);
 
   //TODO: Get the contact to load from the params and fetch.
   //With useEffect, load the contact when params changes
   //and update contact state
 
+  const { id } = useParams();
+
+  useEffect(() => {
+    fetch(`http://localhost:4000/contacts/${id}`)
+      .then((res) => res.json())
+      .then((data) => {
+        console.log(data);
+        setContact(data);
+      });
+  }, [id]);
+
   if (!contact) {
-    return <p>Loading</p>
+    return <p>Loading</p>;
   }
 
   return (
     <div>
-      <h2>{contact.firstName} {contact.lastName}</h2>
-      <p>{contact.street} {contact.city}</p>
+      <h2>
+        {contact.firstName} {contact.lastName}
+      </h2>
+      <p>
+        {contact.street} {contact.city}
+      </p>
     </div>
-  )
+  );
 }
 
-export default ContactsView
\ No newline at end of file
+export default ContactsView;
```

Comment:
I'm struggling to find where you have implemented the edit functionality?

Reply:
well organised challenge, executed all the necessary database fetches and deletes, readable code :-)